alns/stopping_criteria/tests/test_max_runtime.py

```diff
@@ -0,0 +1,78 @@
+import time
+import pytest
+
+from numpy.testing import assert_, assert_almost_equal, assert_raises
+
+from alns.stopping_criteria import MaxRuntime
+from alns.tests.states import Zero
+
+
+def sleep(duration, get_now=time.perf_counter):
+    """
+    Custom sleep function. Built-in time.sleep function is not precise
+    and has different performances depending on the OS, see
+    https://stackoverflow.com/questions/1133857/how-accurate-is-pythons-time-sleep
+    """
+    now = get_now()
+    end = now + duration
+    while now < end:
+        now = get_now()
+
+
+@pytest.mark.parametrize("max_runtime", [-0.001, -1, -10.1])
+def test_raise_negative_parameters(max_runtime: float):
+    """
+    Maximum runtime may not be negative.
+    """
+    with assert_raises(ValueError):
+        MaxRuntime(max_runtime)
+
+
+@pytest.mark.parametrize("max_runtime", [0.001, 1, 10.1])
+def test_valid_parameters(max_runtime: float):
+    """
+    Does not raise for non-negative parameters.
+    """
+    MaxRuntime(max_runtime)
+
+
+@pytest.mark.parametrize("max_runtime", [0.01, 0.1, 1])
+def test_max_runtime(max_runtime):
+    """
+    Test if the max time parameter is correctly set.
+    """
+    stop = MaxRuntime(max_runtime)
+    assert_(stop.max_runtime, max_runtime)
+
+
+@pytest.mark.parametrize("target_runtime", [0.1, 0.5, 1, 2])
```

Comment:
Issue: one and two seconds is much too long for a test. I would prefer `0.01`, `0.05`, and `0.1` (if those work), so as not to make the test suite take unnecessarily long.

Reply:
See [f02aa1e](https://github.com/N-Wouda/ALNS/pull/64/commits/f02aa1e094e4768b6a4808f91500b18c480221eb)